catroid/src/main/java/org/catrobat/catroid/transfers/LoginTask.java

```diff
@@ -64,61 +66,64 @@ public void setOnLoginListener(OnLoginListener listener) {
 	@Override
 	protected void onPreExecute() {
 		super.onPreExecute();
-		if (context == null) {
+		if (contextWeakReference.get() == null) {
```

Comment:
Please read up on the correct usage of `WeakReference`. `get()` returns a regular reference, true, but unless you save this returned value somewhere, the `WeakReference` you hold does not prevent the garbage collector from collecting the context your weak reference points to. So you have to introduce a new local var, write the result of `contextWeakReference.get()` to it and use this regular reference for the remainder of this method. 

Otherwise you risk running into a NPE

Reply:
Thanks!